src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1833,6 +1833,7 @@ private TypeParameterConstraintClauseSyntax ParseTypeParameterConstraintClause()
                     while (true)
                     {
                         if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken
+                            || this.CurrentToken.Kind == SyntaxKind.SemicolonToken
```

Comment:
Hm, I just realized a problem with this change. It allows `abstract class C<T> where T : class;`.
That's not allowed by the [spec](https://github.com/dotnet/csharplang/blob/master/spec/classes.md#class-body).

Should we allow for consistency, or keep disallowing?

Reply:
> Should we allow for consistency, or keep disallowing?

I think the ```;``` should only terminate record.

---
In reply to: [454688376](https://github.com/dotnet/roslyn/pull/45943#discussion_r454688376) [](ancestors = 454688376)